vscode/src/webview/components/App.tsx

```diff
@@ -1,35 +1,69 @@
-import * as React from "react";
+import React, { useState, useEffect } from "react";
+import { Incident } from "../types";
+import { vscode } from "../globals";
+import {
+  Page,
+  PageSection,
+  PageSectionVariants,
+  Title,
+  EmptyState,
+  EmptyStateIcon,
+  EmptyStateBody,
+  Button,
+} from "@patternfly/react-core";
+import { SearchIcon } from "@patternfly/react-icons";
+import IncidentList from "./IncidentList";
 
-// Global variable vscode pulled from the window object.
-interface vscode {
-  postMessage(message: any): void;
-}
-declare const vscode: vscode;
+const App: React.FC = () => {
+  const [incidents, setIncidents] = useState<Incident[]>([]);
 
-const sendMessage = () => {
-  console.log("button clicked");
-  vscode.postMessage({ command: "testing" });
-};
-
-const App = () => {
-  const [buttonText, setButtonText] = React.useState("The brain is pending");
+  useEffect(() => {
+    vscode.postMessage({ command: "requestIncidentData" });
 
-  React.useEffect(() => {
-    window.addEventListener("message", (event) => {
-      const message = event.data; // The json data that the extension sent
+    const handleMessage = (event: MessageEvent) => {
+      const message = event.data;
       switch (message.command) {
-        case "refactor":
-          setButtonText("The brain is working");
+        case "incidentData":
+          if (message.data && message.data.length > 0) {
+            setIncidents(message.data);
+          }
           break;
+        // Handle other messages...
       }
-    });
-  });
+    };
+
+    window.addEventListener("message", handleMessage);
+
+    return () => {
+      window.removeEventListener("message", handleMessage);
+    };
+  }, []);
 
   return (
-    <div>
-      <h1>Functional Components Work!</h1>
-      <button onClick={sendMessage}>{buttonText}</button>
-    </div>
+    <Page>
+      <PageSection variant={PageSectionVariants.light}>
+        <Title headingLevel="h1">Konveyor Analysis Results</Title>
+      </PageSection>
+      <PageSection>
+        {incidents.length > 0 ? (
+          <IncidentList incidents={incidents} />
+        ) : (
+          <EmptyState>
+            <EmptyStateIcon icon={SearchIcon} />
+            <Title headingLevel="h4" size="lg">
+              No Incidents Found
+            </Title>
+            <EmptyStateBody>Run an analysis to see results here.</EmptyStateBody>
+            <Button
+              variant="primary"
+              onClick={() => vscode.postMessage({ command: "startAnalysis" })}
```

Comment:
What's happening here. I now notice it all over the place and in the cat coding webview example. Is this actually running the command or doing something else?

Reply:
Ah it is just passing back a block of data from webview to extension listener... the command term is probably overloaded here.